pkg/network/infraconfigurators/bridge_test.go

```diff
@@ -158,4 +158,484 @@ var _ = Describe("Bridge infrastructure configurator", func() {
+					handler.EXPECT().LinkByName(generateDummyIfaceName(ifaceName)).Return(podLinkAfterNameChange, nil)
+					handler.EXPECT().LinkAdd(dummySwap).Return(nil)
+					handler.EXPECT().AddrReplace(dummySwap, &podIP).Return(nil)
+					handler.EXPECT().SetRandomMac(generateDummyIfaceName(ifaceName)).Return(mac, nil)
+					handler.EXPECT().LinkAdd(inPodBridge).Return(nil)
+					handler.EXPECT().LinkSetMaster(podLinkAfterNameChange, inPodBridge).Return(nil)
+					handler.EXPECT().LinkSetUp(inPodBridge).Return(nil)
+					handler.EXPECT().ParseAddr(bridgeIPStr).Return(bridgeIPAddr, nil)
+					handler.EXPECT().AddrAdd(inPodBridge, bridgeIPAddr).Return(nil)
+					handler.EXPECT().DisableTXOffloadChecksum(inPodBridge.Name).Return(nil)
+					handler.EXPECT().CreateTapDevice(
+						tapDeviceName, queueCount, launcherPID, mtu, netdriver.LibvirtUserAndGroupId).Return(nil)
+					handler.EXPECT().BindTapDeviceToBridge(tapDeviceName, inPodBridge.Name).Return(nil)
+					handler.EXPECT().ConfigureIpv4ArpIgnore().Return(nil)
+					handler.EXPECT().LinkSetUp(podLinkAfterNameChange).Return(fmt.Errorf("failed to set pod link UP"))
+				})
+
+				It("should fail", func() {
+					Expect(bridgeConfigurator.PreparePodNetworkInterface()).To(HaveOccurred())
+				})
+			})
+
+			When("we fail to set the pod link learning off", func() {
+				BeforeEach(func() {
+					handler.EXPECT().LinkSetDown(podLink).Return(nil)
+					handler.EXPECT().AddrDel(podLink, &podIP).Return(nil)
+					handler.EXPECT().LinkSetName(podLink, generateDummyIfaceName(ifaceName)).Return(nil)
+					handler.EXPECT().LinkByName(generateDummyIfaceName(ifaceName)).Return(podLinkAfterNameChange, nil)
+					handler.EXPECT().LinkAdd(dummySwap).Return(nil)
+					handler.EXPECT().AddrReplace(dummySwap, &podIP).Return(nil)
+					handler.EXPECT().SetRandomMac(generateDummyIfaceName(ifaceName)).Return(mac, nil)
+					handler.EXPECT().LinkAdd(inPodBridge).Return(nil)
+					handler.EXPECT().LinkSetMaster(podLinkAfterNameChange, inPodBridge).Return(nil)
+					handler.EXPECT().LinkSetUp(inPodBridge).Return(nil)
+					handler.EXPECT().ParseAddr(bridgeIPStr).Return(bridgeIPAddr, nil)
+					handler.EXPECT().AddrAdd(inPodBridge, bridgeIPAddr).Return(nil)
+					handler.EXPECT().DisableTXOffloadChecksum(inPodBridge.Name).Return(nil)
+					handler.EXPECT().CreateTapDevice(
+						tapDeviceName, queueCount, launcherPID, mtu, netdriver.LibvirtUserAndGroupId).Return(nil)
+					handler.EXPECT().BindTapDeviceToBridge(tapDeviceName, inPodBridge.Name).Return(nil)
+					handler.EXPECT().ConfigureIpv4ArpIgnore().Return(nil)
+					handler.EXPECT().LinkSetUp(podLinkAfterNameChange).Return(nil)
+					handler.EXPECT().LinkSetLearningOff(podLinkAfterNameChange).Return(fmt.Errorf("failed to set link learning off"))
+				})
+
+				It("should fail", func() {
+					Expect(bridgeConfigurator.PreparePodNetworkInterface()).To(HaveOccurred())
+				})
+			})
+		})
+
+		When("the pod features an L2 network (no IPAM)", func() {
```

Comment:
ditto

Reply:
I honestly don't know what you're referring to, sorry.

Reply:
nevermind